sw/device/lib/runtime/ibex.h

```diff
@@ -27,15 +27,28 @@ extern const size_t kIbexClockFreqHz;
  *
  * @param cycles the number of cycles to burn.
  */
-inline void ibex_busy_loop(size_t cycles) {
-  size_t out;  // Used to create an inout parameter below.
+inline void ibex_busy_loop(uint64_t cycles) {
+  // Round down |cycles| to closest multiple of 8
+  cycles = cycles & ~7ULL;
+  // Unpack uint64_t into two uint32_ts.
+  uint32_t cycles_lower = cycles;
+  uint32_t cycles_upper = cycles >> 32;
+  // To tell the compiler that |cycles_lower| and |cycles_upper| are modified.
+  uint32_t out_lower, out_upper;
+  // Takes approximately 8 cycles per interation
```

Comment:
Shouldn't it be exactly 8 cycles?

Reply:
Ideally, yes. I wrote _approximately_ because of the outer loop (labelled `upper`), which runs once per `UINT32_MAX` cycles. I can add more details or remove the comment if it is confusing/misleading.

Reply:
Using the cycle counter resolves this issue.